src/main/java/no/nav/folketrygdloven/beregningsgrunnlag/arbeidstaker/BeregnPrArbeidsforholdFraAOrdningen.java

```diff
@@ -38,13 +43,53 @@ public Evaluation evaluate(BeregningsgrunnlagPeriode grunnlag) {
 
         List<BigDecimal> inntekter = grunnlag.getInntektsgrunnlag().getPeriodeinntekter(Inntektskilde.INNTEKTSKOMPONENTEN_BEREGNING, arbeidsforhold, bp.getTom(), beregningsPeriodeLengdeIMnd);
         BigDecimal sum = inntekter.stream().reduce(BigDecimal::add).orElse(BigDecimal.ZERO);
-        BigDecimal beregnetPrÅr = sum.multiply(BigDecimal.valueOf(12)).divide(BigDecimal.valueOf(beregningsPeriodeLengdeIMnd), 10, RoundingMode.HALF_EVEN);
+        BigDecimal snittFraBeregningsperiodenPrÅr = sum.multiply(BigDecimal.valueOf(12)).divide(BigDecimal.valueOf(beregningsPeriodeLengdeIMnd), 10, RoundingMode.HALF_EVEN);
+        BigDecimal andelFraAOrdningenPrÅr = finnAndelAvBeregnet(snittFraBeregningsperiodenPrÅr, arbeidsforhold, grunnlag);
         BeregningsgrunnlagPrArbeidsforhold.builder(arbeidsforhold)
-            .medBeregnetPrÅr(beregnetPrÅr)
+            .medBeregnetPrÅr(andelFraAOrdningenPrÅr)
             .build();
         Map<String, Object> resultater = new HashMap<>();
         resultater.put("beregnetPrÅr", arbeidsforhold.getBeregnetPrÅr());
         resultater.put("arbeidsforhold", arbeidsforhold.getBeskrivelse());
         return beregnet(resultater);
     }
+
+    private BigDecimal finnAndelAvBeregnet(BigDecimal beregnetPrÅr, BeregningsgrunnlagPrArbeidsforhold arbeidsforhold, BeregningsgrunnlagPeriode periode) {
+        YtelsesSpesifiktGrunnlag ytelsesSpesifiktGrunnlag = periode.getBeregningsgrunnlag().getYtelsesSpesifiktGrunnlag();
+        if (!arbeidsforhold.erFrilanser() && ytelsesSpesifiktGrunnlag instanceof OmsorgspengerGrunnlag) {
+            return fordelRestinntektFraAOrdningen(beregnetPrÅr, arbeidsforhold, periode);
+        }
+        return beregnetPrÅr;
+    }
+
+    private BigDecimal fordelRestinntektFraAOrdningen(BigDecimal beregnetPrÅr, BeregningsgrunnlagPrArbeidsforhold arbeidsforhold, BeregningsgrunnlagPeriode periode) {
+        Inntektsgrunnlag inntektsgrunnlag = periode.getInntektsgrunnlag();
+        List<BeregningsgrunnlagPrArbeidsforhold> arbeidsforholdISammeOrg = periode.getBeregningsgrunnlagPrStatus(AktivitetStatus.ATFL)
+            .getArbeidsforholdIkkeFrilans()
+            .stream()
+            .filter(a -> a.getArbeidsgiverId().equals(arbeidsforhold.getArbeidsgiverId()))
```

Comment:
Kan man risikere at andeler uten arbeidsgiver kommer inn her? Eks etterlønn/sluttpakke andeler? De vil isåfall gi NPE her. Litt usikker

Reply:
Godt poeng. Legger inn sjekk.